packages/xdl/src/detach/AndroidShellApp.js

```diff
@@ -1040,24 +1041,83 @@ export async function runShellAppModificationsAsync(
   );
 }
 
-async function buildShellAppAsync(context: StandaloneContext, sdkVersion: string) {
+async function buildShellAppAsync(
+  context: StandaloneContext,
+  sdkVersion: string,
+  buildType: string
+) {
   let shellPath = shellPathForContext(context);
+  const ext = buildType === 'app-bundle' ? 'aab' : 'apk';
 
-  if (context.build.android) {
-    let androidBuildConfiguration = context.build.android;
+  const isRelease = !!context.build.android;
+  // concat on those strings is not very readable, but only alternative here is huge if statement
+  const debugOrRelease = isRelease ? 'Release' : 'Debug';
+  const devOrProd = isRelease ? 'Prod' : 'Dev';
+  const debugOrReleaseL = isRelease ? 'release' : 'debug';
+  const devOrProdL = isRelease ? 'prod' : 'dev';
 
-    try {
-      await fs.remove(`shell-unaligned.apk`);
-      await fs.remove(`shell.apk`);
-    } catch (e) {}
-    let gradleBuildCommand;
+  const shellFile = `shell.${ext}`;
+  const shellUnalignedFile = `shell-unaligned.${ext}`;
+
+  const outputDirPath = path.join(
+    shellPath,
+    'app',
+    'build',
+    'outputs',
+    buildType === 'app-bundle' ? 'bundle' : 'apk'
+  );
+
+  let gradleBuildCommand;
+  let outputPath;
+  if (buildType === 'app-bundle') {
+    if (ExponentTools.parseSdkMajorVersion(sdkVersion) >= 33) {
+      gradleBuildCommand = `bundle${debugOrRelease}`;
+      outputPath = path.join(outputDirPath, debugOrReleaseL, `app.aab`);
+    } else if (ExponentTools.parseSdkMajorVersion(sdkVersion) >= 32) {
+      gradleBuildCommand = `bundle${devOrProd}Kernel${debugOrRelease}`;
+      outputPath = path.join(outputDirPath, `${devOrProdL}Kernel${debugOrRelease}`, `app.aab`);
```

Comment:
So we do support SDK32?

Reply:
yes 32 and 33 for app bundles (this has nothing to do width 64bit support)
older sdk should also work, but there were some problems with the build and I don't think it's worth investigating given that it will need to be deprecated when 64 bit builds will be required.